src/github.com/matrix-org/dendrite/clientapi/storage/syncserver.go

```diff
@@ -86,8 +88,12 @@ func (d *SyncServerDatabase) SetPartitionOffset(topic string, partition int32, o
 }
 
 // SyncStreamPosition returns the latest position in the sync stream. Returns 0 if there are no events yet.
-func (d *SyncServerDatabase) SyncStreamPosition() (int64, error) {
-	return d.events.MaxID()
+func (d *SyncServerDatabase) SyncStreamPosition() (syncapi.StreamPosition, error) {
```

Comment:
how about making `EventsInRange` take a `StreamPosition`? I think that would be consistent?